From 3b1aeb5e92f9d175520cd581bbeec2043570a9fc Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:33:32 -0400
Subject: [alsa] hda: Add fixup_forced flag

Message-id: <140972823573-312-patchset-jkysela@redhat.com>
Patchwork-id: 91008
O-Subject: [RHEL 7.1 PATCH 311/442] ALSA: hda - Add fixup_forced flag
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

The "fixup_forced" flag will indicate whether a specific fixup
(or nofixup) has been set by the user, to override the driver's
default.
This flag will help future patches.

Signed-off-by: David Henningsson <david.henningsson@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: c21c8cf77f47e234b5724d24dd133dc034351327)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_auto_parser.c b/sound/pci/hda/hda_auto_parser.c
index 90d2fda..36961ab 100644
--- a/sound/pci/hda/hda_auto_parser.c
+++ b/sound/pci/hda/hda_auto_parser.c
@@ -852,15 +852,17 @@ void snd_hda_pick_fixup(struct hda_codec *codec,
  if (codec->modelname && !strcmp(codec->modelname, "nofixup")) {
   codec->fixup_list = NULL;
   codec->fixup_id = -1;
+  codec->fixup_forced = 1;
   return;
  }
 
  if (codec->modelname && models) {
   while (models->name) {
    if (!strcmp(codec->modelname, models->name)) {
-    id = models->id;
-    name = models->name;
-    break;
+    codec->fixup_id = models->id;
+    codec->fixup_name = models->name;
+    codec->fixup_forced = 1;
+    return;
    }
    models++;
   }
@@ -889,6 +891,7 @@ void snd_hda_pick_fixup(struct hda_codec *codec,
   }
  }
 
+ codec->fixup_forced = 0;
  codec->fixup_id = id;
  if (id >= 0) {
   codec->fixup_list = fixlist;
diff --git a/sound/pci/hda/hda_codec.h b/sound/pci/hda/hda_codec.h
index a423313..5825aa1 100644
--- a/sound/pci/hda/hda_codec.h
+++ b/sound/pci/hda/hda_codec.h
@@ -402,6 +402,7 @@ struct hda_codec {
 
  /* fix-up list */
  int fixup_id;
+ unsigned int fixup_forced:1; /* fixup explicitly set by user */
  const struct hda_fixup *fixup_list;
  const char *fixup_name;
 
-- 
1.7.1